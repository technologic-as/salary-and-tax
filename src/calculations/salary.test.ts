import {
  calculateSurplus,
  defaultSalaryParameters,
  getCompanyIncome,
  getEmployerFee,
  getSalaryCalculations,
  getTurnover,
  getVacationSavings,
} from './salary';

const data = defaultSalaryParameters;
const expectedTurnover = 1903000;
const expectedAfterCut = expectedTurnover * 0.9;
const expectedAfterEmployerFee = 1501052;
const expectedAfterVacationSavings = 1334269;

describe('salary', () => {
  describe('getSalaryCalculations', () => {
    it('should match snapshot', () => {
      expect(getSalaryCalculations(data)).toMatchSnapshot();
    });
  });
  describe('getTurnover', () => {
    const hoursPerYear = '1730';
    const hourRate = '1100';
    it('should calculate turnover and return object', function () {
      expect(getTurnover({ hoursPerYear, hourRate })).toEqual({
        turnover: expectedTurnover,
      });
    });
  });
  describe('getCompanyIncome', () => {
    it('should calculate based on salary and return object', () => {
      expect(getCompanyIncome(expectedTurnover, { cut: 10 })).toEqual({
        after: expectedAfterCut,
        theCut: expectedTurnover * 0.1,
      });
    });
  });
  describe('calculateSurplus', () => {
    const income = 100000;
    const amount = 1000;
    it('should calculate surplus when true', () => {
      expect(calculateSurplus(income, { surplus: { amount, include: true } })).toEqual({
        after: income - amount,
        surplus: amount,
      });
    });
    it('should not calculate surplus when false', () => {
      expect(
        calculateSurplus(income, {
          surplus: { amount, include: false },
        })
      ).toEqual({ after: income, surplus: 0 });
    });
  });
  describe('getEmployerFee', () => {
    it('should calculate the employer fee when included', () => {
      expect(
        getEmployerFee(expectedAfterCut, {
          employerFee: { include: true, rate: 14.1 },
        })
      ).toEqual({ after: expectedAfterEmployerFee, employerFee: 211648 });
    });
    it('should not calculate the employer fee when not included', () => {
      expect(
        getEmployerFee(expectedAfterCut, {
          employerFee: { include: false, rate: 14.1 },
        })
      ).toEqual({ after: expectedAfterCut, employerFee: 0 });
    });
  });
  describe('getVacationSavings', () => {
    it('should calculate the vacation savings when included', () => {
      expect(
        getVacationSavings(expectedAfterEmployerFee, {
          vacationSavings: { include: true, rate: 12.5 },
        })
      ).toEqual({
        after: expectedAfterVacationSavings,
        vacationSavings: 166783,
      });
    });
    it('should not calculate the vacation savings when not included', () => {
      expect(
        getVacationSavings(expectedAfterEmployerFee, {
          vacationSavings: { include: false, rate: 12.5 },
        })
      ).toEqual({ after: expectedAfterEmployerFee, vacationSavings: 0 });
    });
  });
});
