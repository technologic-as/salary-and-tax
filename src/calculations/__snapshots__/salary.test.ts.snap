// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`salary getSalaryCalculations should match snapshot 1`] = `
Object {
  "companyIncome": 1712700,
  "employerFee": 211648,
  "income": 1315701,
  "pension": 24524,
  "surplus": 0,
  "theCut": 190300,
  "totalCuts": 587299,
  "turnover": 1903000,
  "vacationSavings": 160827,
  "withoutEmployerFee": 1501052,
  "withoutPension": 1315701,
  "withoutSurplus": 1712700,
  "withoutVacationSavings": 1340225,
}
`;
